pennylane/tape/qscript.py

```diff
@@ -695,7 +695,16 @@ def set_parameters(self, params, trainable_only=True):
 
         for idx, p in iterator:
             op = self._par_info[idx]["op"]
-            op.data[self._par_info[idx]["p_idx"]] = p
+
+            if isinstance(
+                op, Tensor
+            ):  # the setter for tensor does not support the definition syntax
```

Comment:
what does this mean? will Tensors still hit the bug you're trying to fix?

Reply:
More like: the old syntax, which is buggy for operator arithmetic, is the only thing that works for Tensor. 😢

Reply:
Do `Tensor`'s even ever have data? I guess if they contain a `Hermitian` or `Projector`.